packages/gitgraph-core/src/gitgraph.ts

```diff
@@ -289,29 +291,27 @@ export class GitgraphCore {
   /**
    * Import a JSON.
    *
-   * Note: because data come from a json, we can't be sure about the type
-   * of this one, it's why `data` is `any` here, and will be validate by the function
+   * Data can't be typed since it comes from a JSON.
+   * We validate input format and throw early if something is invalid.
    *
    * @param data JSON from `git2json` output
    */
   public import(data: any) {
-    // TODO: validate data format with https://github.com/hapijs/joi
-    this.clear();
+    // Validate `data` format.
+    const schema = Joi.array().items(
+      Joi.object({
+        refs: Joi.array(),
```

Comment:
Nice 👍 We also need to specify all others params of commit by it's a good start :)

Reply:
I suggest we only validate what we actually use, what do you think?

I set `allowUnknown: true` so it won't fail if there are attributes that are not restricted here. This way, it won't break for eventual properties that won't be used here.

Reply:
Sure, but we are spreading `commit` just after, so we need to have the correct signature ;) (we are actually using almost all fields except `tree` in my memory)